### PR TITLE
[YUNIKORN-2205] remove the warning of processing nonexistent "namespa…

### PR DESCRIPTION
**Pull request by Haser0305**
### What is this PR for?
The warning message is printed when "namespace.guaranteed" does not exist. This PR checking if the "namespaceGuaranteed" value is empty and returning early to avoid the warning message. This ensures that the warning is not displayed when the expected behavior occurs. 

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2205

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

**Review by pbacsko — CHANGES_REQUESTED** [review]
Nit: please add coverage in `utils_test.go`

File: pkg/common/utils/utils.go [hunk 1]
```diff
@@ -180,6 +180,9 @@ func PodUnderCondition(pod *v1.Pod, condition *v1.PodCondition) bool {
 func GetNamespaceGuaranteedFromAnnotation(namespaceObj *v1.Namespace) *si.Resource {
 	// retrieve guaranteed resource info from annotations
 	namespaceGuaranteed := GetNameSpaceAnnotationValue(namespaceObj, constants.NamespaceGuaranteed)
```
Comment (pbacsko) [inline pkg/common/utils/utils.go:182]: I believe this is more Go-like:
```
if guaranteed := GetNameSpaceAnnotationValue(namespaceObj, constants.NamespaceGuaranteed); guaranteed != "" {
  ...
}
return nil
```

No need to name it "namespaceGuaranteed", it's pretty obvious from the context. Prefer shorter variables if possible.

Comment (Haser0305) [reply pkg/common/utils/utils.go:182]: Thanks for reviewing my code and the helpful recommendation.  I'll make that change to implement your feedback.

**Review by zhuqi-lucas — APPROVED** [review]
LGTM

**Pull request closed without merging**